File: scan_and_go_flutter/lib/models/product.dart
class Product{
  const Product({
    required this.id,
    required this.name,
    required this.price,
    required this.weight,
  });

  final int id;
  final String name;
  final double price;
  final double weight;
  
  Map<String, dynamic> toJson() {
    return {
      'Id': id,
      'Name': name,
      'Price': price,
      'Weight': weight,
    };
  }

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['Id'],
      name: json['Name'],
      price: json['Price'] is int ? (json['Price'] as int).toDouble() : json['Price'],
      weight: json['Weight'] is int ? (json['Weight'] as int).toDouble() : json['Weight'],
    );
  }
}

File: ScanAndGo/ScanAndGoMVC/ModelsLibrary/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace ModelsLibrary.Models
{
    public class Product
    {
        [Key]
        public int Id { get; private set; }
        [Required]
        public string Name { get; private set; }

        public double Price { get; private set; }
        public double Weight { get; private set; }

        public int? StoreId { get; private set; }

        [ForeignKey("StoreId")]
        public  Store? Store { get; private set; }

        [StringLength(20)]
        public string? Barcode { get; private set; }

        private Product(int id, string name, double price, double weight, int? storeId, string? barcode)
        {
            Id = id;
            Name = name;
            Price = price;
            Weight = weight;
            StoreId = storeId;
            Barcode = barcode;
        }

        public void Update(string name, double price, double weight, string barcode, int? storeId)
        {
            Name = name;
            Price = price;
            Weight = weight;
            Barcode = barcode;
            StoreId = storeId;
        }

        public static Product Create(int id, string name, double price, double weight, int storeId, string barcode)
        {
            return new Product(id, name, price, weight, storeId, barcode);
        }
    }
}
